src/main/java/org/opentripplanner/middleware/otp/OtpDispatcher.java

```diff
@@ -56,19 +82,14 @@ private static OtpDispatcherResponse sendOtpRequest(URI uri) {
                 .GET()
                 .build();
 
+        // Get response from OTP
         OtpDispatcherResponse otpDispatcherResponse = null;
-
-        // get response from OTP
         try {
             HttpResponse<String> otpResponse = client.send(request, HttpResponse.BodyHandlers.ofString());
-            otpDispatcherResponse = new OtpDispatcherResponse();
-            otpDispatcherResponse.responseBody = otpResponse.body();
-            otpDispatcherResponse.statusCode = otpResponse.statusCode();
-            LOG.debug("Response from OTP server: {}", otpDispatcherResponse.toString());
+            otpDispatcherResponse = new OtpDispatcherResponse(otpResponse);
         } catch (InterruptedException | IOException e) {
             LOG.error("Error requesting OTP data", e);
         }
-
         return otpDispatcherResponse;
```

Comment:
This method can return null which can result in NPEs. I would prefer for this method to throw a non-runtime exception if the request failed so that code that uses this method can properly account for times when an OTP request fails.

Reply:
This seems out of scope for this PR. Will create an issue for this.